phoenix-core/src/it/java/org/apache/phoenix/end2end/index/ViewIndexIT.java

```diff
@@ -280,6 +282,64 @@ public void testCoprocsOnGlobalNonMTImmutableViewIndex() throws Exception {
         testCoprocsOnGlobalViewIndexHelper(false, false);
     }
 
+    @Test
+    public void testDroppingColumnWhileCreatingIndex() throws Exception {
+        String schemaName = "S1";
+        String tableName = generateUniqueName();
+        String viewSchemaName = "S1";
+        String fullTableName = SchemaUtil.getTableName(schemaName, tableName);
+        String indexName = "IND_" + generateUniqueName();
+        String viewName = "VIEW_" + generateUniqueName();
+        String fullViewName = SchemaUtil.getTableName(viewSchemaName, viewName);
+
+        createBaseTable(schemaName, tableName, false, null, null, true);
+        try (Connection conn = getConnection()) {
+            conn.setAutoCommit(true);
+            conn.createStatement().execute("CREATE VIEW " + fullViewName + " AS SELECT * FROM " + fullTableName);
+            conn.commit();
+
+            final int[] exceptionCode = new int[1];
```

Comment:
Not a strong opinion but since we want threads to mutate exceptionCode, maybe we can use `AtomicInteger` for exceptionCode rather than array with size 1, threads can set values and finally we can assert with `assertEquals(exceptionCode.get(), 301)` if you would prefer :)

Reply:
`exceptionCode` is accessed from an inner class so has to be final, in order to be able to change its value from the other thread it has to be an array.
We can use AtomicInteger array but I am not sure if it adds any value.

Reply:
I meant AtomicInteger as mutable final object similar to an array.
e.g
```
            final AtomicInteger exceptionCode = new AtomicInteger();

```
&
```
                    } catch (SQLException e) {
                        exceptionCode.set(e.getErrorCode());
                        throw new RuntimeException(e);
                    } finally {
                        doneSignal.countDown();
                    }
```
&
```
            assertEquals(exceptionCode.get(), 301);
```
Thought?

Reply:
Yeah I think this a better solution. Nice one :) 
Thanks for the review @virajjasani I have updated the commit.